File: ApiProject/API-TFG/src/main/java/com/aitor/api_tfg/model/db/Comment.java
package com.aitor.api_tfg.model.db;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "comments")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "publication_id", nullable = false)
    private Publication publication;

    @Column(nullable = false, length = 250)
    private String text;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime creationDate;
}

File: AndroidProject/app/src/main/java/com/aitor/trackactividades/feed/presentation/model/Comment.kt
package com.aitor.trackactividades.feed.presentation.model

import java.time.LocalDateTime

data class Comment(
    val id: Long,
    val userId: Int,
    val publicationId: Long,
    val comment: String,
    val userName: String,
    val userImage: String,
    val creationDate: LocalDateTime
)
